mmcv/ops/csrc/ms_deform_attn_cuda_kernel.cuh

```diff
@@ -0,0 +1,810 @@
+/*!
+**************************************************************************************************
+* Deformable DETR
+* Copyright (c) 2020 SenseTime. All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 [see LICENSE for details]
+**************************************************************************************************
+* Modified from
+*https://github.com/chengdazhi/Deformable-Convolution-V2-PyTorch/tree/pytorch_1.0.0
+**************************************************************************************************
+*/
+#ifndef OPENMMLAB_PRS_ADD_MSD_ATTEN_MMCV_OPS_CSRC_MS_DEFORM_ATTN_CUDA_KERNEL_CUH_
+#define OPENMMLAB_PRS_ADD_MSD_ATTEN_MMCV_OPS_CSRC_MS_DEFORM_ATTN_CUDA_KERNEL_CUH_
+
+#include "pytorch_cuda_helper.hpp"
+
+#define CUDA_KERNEL_LOOP(i, n)                                 \
```

Comment:
The "common_cuda_helper" already implements some functions like `CUDA_KERNEL_LOOP`, see https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/common_cuda_helper.hpp